### PR TITLE
iterates through data, makes predictions using the 3 model

### LINKED ISSUE #9: Test on new test set
Could someone please:

- [x] Add the test set provided to us
- [x] Write logic to load this in
- [ ] Test on this test set (NLTK has a POS tagger. Please use this as a "ground truth" that we measure our accuracy against). This is probably the best we have at measuring our accuracy since we don't have labels.

There is now a PR linked to this, updating what has been done by checking the boxes

### PR DESCRIPTION
**Pull request by Kofwana**
This updated code will create a file named "NLP_Pioneers.test.txt" with two columns (token and predicted_tag). It iterates through the test data, makes predictions using the three models (Bayes, SVM, Logistic Regression), and writes the token and predicted tag to the output file. Let me know if anything stands out regarding what I've implemented so far.

### CODE REVIEW

File: main.py [hunk 1]
```diff
@@ -1,17 +1,16 @@
...
-from algos.bayes import Bayes 
+from algos.bayes import Bayes
 from algos.utils import get_data
 
-if __name__=="__main__":
-
+if __name__ == "__main__":
     # Get the data and split
-    train_data, test_data = get_data() 
+    train_data, test_data = get_data()
```
Comment (NoahSchiro) [inline main.py]: This is broken now because get_data returns three objects. The train, the val, and the test. Since val isn't used anywhere in this file, this line needs to be changed to

train_data, _, test_data = get_data()

File: main.py [hunk 2]
```diff
@@ -1,17 +1,16 @@
...
 
     # Prepare the data for training and testing
     train_features, train_labels = utils.prepare_data(train_data)
-    test_features, test_labels   = utils.prepare_data(test_data)
+    test_features, test_labels = utils.prepare_data(test_data)
```
Comment (NoahSchiro) [inline main.py]: If we want to have that validate now, then we need to include it here as well (but it doesn't look like we use it anywhere in the rest of the file

File: main.py [hunk 3]
```diff
@@ -26,44 +25,45 @@
...
-    for (feature, label) in zip(test_features, test_labels):
-
-        # Make a prediction for bayes
-        b_pred = bayes.forward(feature)
-
-        # Make a prediction for svm
-        s_pred = svm.predict_pos_tags([feature["token"]], svm_vectorizer, svm_model)
+    # Open the output file for writing
+    with open("NLP_Pioneers.test.txt", "w") as output_file:
+        for (feature, label) in zip(test_features, test_labels):
```
Comment (NoahSchiro) [inline main.py]: there are no test labels we have to just guess the POS tag given their token.

File: main.py [hunk 4]
```diff
@@ -26,44 +25,45 @@
...
+                pred = s_pred
 
-        # Okay we have our prediction, let's see how our accuracy goes
+            # If NONE of them equal each other, then just use the model
+            # with the highest accuracy
+            else:
+                pred = l_pred
 
-        total += 1
-        if pred == label:
```
Comment (NoahSchiro) [inline main.py:64]: There is no label and therefore no way for us to measure / report accuracy here.

**Review by NoahSchiro — CHANGES_REQUESTED** [review]

**Comment by NoahSchiro** [comment]
Kofwana this looks great. One comment: the output is something along the lines of ['NNP'] or whatever the POS is. Could we change this to simply be NNP so that it looks similar to the training input data?

Secondly, I am not sure I was right about the comma delimited output.... If you look at the training data it appears to be space delimited. Could we switch to this instead? Sorry for the confusion

**Review by NoahSchiro — APPROVED** [review]

**Review by Kofwana — COMMENTED** [review]
The output is now in the space-delimited format similar to the training input data, and the predicted POS tags will be without square brackets to match the format.

**Review by afq5024 — APPROVED** [review]
